actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -112,27 +115,72 @@ private void addCourseDetails(Request request, Map<String, Object> userCoursesRe
             requestBody,
             (Map<String, String>) request.getRequest().get(JsonKey.HEADER));
 
-    mergeDetailsAndSendCourses(contents, batches);
+    Map<String, Object> courseBatchesMap = null;
+    List<String> requiredMetadata = null;
+    Map<String, String[]> queryParams =
+        (Map<String, String[]>) request.getContext().get(JsonKey.REQUEST_PARAMS);
+
+    if (!MapUtils.isEmpty(queryParams) && queryParams.containsKey(JsonKey.FIELDS)) {
+      List<String> courseBatchIds =
+          (List<String>)
+              batches
+                  .stream()
+                  .map(batch -> (String) batch.get(JsonKey.BATCH_ID))
+                  .collect(Collectors.toSet());
+
+      requiredMetadata = Arrays.asList(queryParams.get(JsonKey.FIELDS));
+      Map<String, Object> esQueryMap = new HashMap<>();
+      esQueryMap.put(JsonKey.IDENTIFIER, courseBatchIds);
+      SearchDTO dto = new SearchDTO();
+      requiredMetadata.add(JsonKey.IDENTIFIER);
+      dto.setFields(requiredMetadata);
+      dto.getAdditionalProperties().put(JsonKey.FILTERS, esQueryMap);
+      courseBatchesMap =
+          ElasticSearchUtil.complexSearch(
+              dto,
+              ProjectUtil.EsIndex.sunbird.getIndexName(),
+              ProjectUtil.EsType.course.getTypeName());
+    }
+
+    Map<String, Object> courseBatches =
+        getContentAsMap((Map<String, Object>) courseBatchesMap.get(JsonKey.CONTENT));
+
+    mergeDetailsAndSendCourses(contents, batches, courseBatches); // metadata
   }
 
   public void mergeDetailsAndSendCourses(
-      Map<String, Object> coursesContents, List<Map<String, Object>> batches) {
+      Map<String, Object> coursesContents,
+      List<Map<String, Object>> batches,
+      Map<String, Object> metadata) {
```

Comment:
why is this variable named metadata = we are passing couresBatches result i guess.